hydromt/models/model.py

```diff
@@ -455,6 +453,38 @@ def write_data_catalog(
 
             cat.to_yml(path, root=root)
 
+    def test_equal(self, other: "Model") -> tuple[bool, dict[str, str]]:
```

Comment:
I would add a skip_components argument here like before.

Reply:
Let's add it once we need it. I haven't found any code that uses it.

Reply:
Plugins ? :)
But in a way now by default you check on the component type which was not there before so maybe implementing a skip argument is less needed than it used to be.

Reply:
Perhaps a nice middle way is to only check equality on components that are loaded? Just an idea, if it doesn't work, feel free to ignore.

Reply:
There is no real way to check if a component is loaded, as it is lazily loaded. Whenever `data` is called, it will be loaded.